core/domain/wipeout_service.py

```diff
@@ -407,7 +542,172 @@ def _pseudonymize_models(activity_related_models, pseudonymized_user_id):
             transaction_services.run_in_transaction(
                 _pseudonymize_models,
                 activity_related_models[i:i + MAX_NUMBER_OF_OPS_IN_TRANSACTION],
-                pseudonymized_user_id)
+                pseudonymized_id)
+
+
+def _pseudonymize_col_or_exp_models(
+        pending_deletion_request,
+        activity_category,
+        snapshot_metadata_model_class,
+        rights_snapshot_metadata_model_class,
+        rights_snapshot_content_model_class,
+        commit_log_model_class,
+        commit_log_model_field_name):
+    """Pseudonymize the collection or exploration models for the user with
+    user_id.
+
+    Args:
+        pending_deletion_request: PendingDeletionRequest. The pending deletion
+            request object to be saved in the datastore.
+        activity_category: models.NAMES. The category of the models that are
+            being pseudonymized.
+        snapshot_metadata_model_class:
+            CollectionSnapshotMetadataModel|ExplorationSnapshotMetadataModel.
+            The snapshot metadata model class.
+        rights_snapshot_metadata_model_class:
+            BaseSnapshotMetadataModel. The rights snapshot metadata model class.
+        rights_snapshot_content_model_class:
+            BaseSnapshotContentModel. The rights snapshot content model class.
+        commit_log_model_class:
+            CollectionCommitLogEntryModel|ExplorationCommitLogEntryModel.
+            The commit log model class.
+        commit_log_model_field_name: str. The name of the field holding the
+            activity id in the corresponding commit log model.
+    """
+    user_id = pending_deletion_request.user_id
+
+    snapshot_metadata_models, commit_log_models = (
+        _collect_activity_ids_from_snapshots_and_commit(
+            pending_deletion_request,
+            activity_category,
+            [
+                snapshot_metadata_model_class,
+                rights_snapshot_metadata_model_class
+            ],
+            commit_log_model_class,
+            commit_log_model_field_name
+        )
+    )
+
+    def _pseudonymize_models(col_related_models, pseudonymized_id):
```

Comment:
Let's say this function fails. I imagine an exception would be raised, but some other models may have been changed in previous for loops. In that case, what will happen and how would we fix that scenario?